csrc/device_lower/lower2device.h

```diff
@@ -238,6 +240,17 @@ class GpuLower : public NonCopyable {
     return passes_;
   }
 
+  template <typename... Args>
+  void validate(Val* val, Args... args) {
+    auto sv = simplifyExpr(val);
+    if (sv->isTrue()) {
```

Comment:
We don't need to store in validation `Val*` objects that are simplified to bool True?

Just to be sure, Val* is ok for us as long as it's not evaluated to bool False, that is, we will think about Val with division results as valid, only because the stored value is not bool False?

Reply:
If it is simplified to true, then this condition is always satisfied regardless of the input of the kernel. For this case, we do not want any check in the runtime. For example, we are not interested in checking that `3 < 4` each time we launch the kernel. We know that it is always true, and just skip it. I will update the code mentioning that.

Reply:
added a comment in the code

Reply:
Thank you for explanation!